pkg/proxy/apis/config/validation/validation.go

```diff
@@ -231,7 +230,7 @@ func validateHostPort(input string, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(fldPath, hostIP, "must be a valid IP"))
 	}
 
-	if p, err := strconv.Atoi(port); err != nil {
+	if p, err := netutils.ParsePort(port, false); err != nil {
```

Comment:
this might be sensitive here inside api validation. Does this change semantics in any way?

Reply:
It prevents the port from overflowing